runtime/src/main/java/org/corfudb/runtime/view/workflows/HealNode.java

```diff
@@ -39,11 +41,27 @@ protected UUID sendRequest(@NonNull ManagementClient managementClient) throws Ti
     protected boolean verifyRequest(@NonNull Layout layout) {
         log.info("verifyRequest: {} in {}", this, layout);
         return layout.getAllServers().contains(nodeForWorkflow)
-                && layout.getSegmentsForEndpoint(nodeForWorkflow).size() == 1;
+                && layout.getSegmentsForEndpoint(nodeForWorkflow).size()
+                == layout.getSegments().size();
     }
 
     @Override
     public String toString() {
         return this.getClass().getSimpleName() + " " + nodeForWorkflow;
     }
+
+    @Override
+    protected Optional<ManagementClient> getOrchestrator() {
+        runtime.invalidateLayout();
+        Layout layout = new Layout(runtime.getLayoutView().getLayout());
+        BaseClient baseClient = runtime.getLayoutView().getRuntimeLayout(layout).getBaseClient(nodeForWorkflow);
+        if (baseClient.pingSync()) {
+            log.info("getOrchestrator: orchestrator selected {}", nodeForWorkflow);
+            return Optional.of(runtime.getLayoutView()
```

Comment:
why it doesn't have `sleep` like in AddNode class?

Reply:
AddNode is initiated from the outside the cluster, and could be executed right after a node was removed. HealNode is initiated from the RemoteMonitoringService of the current node and is performed when a node is healed and running for a while.